samples/matter/common/src/binding_handler.cpp

```diff
@@ -27,62 +15,78 @@ using namespace chip::app;
 
 void BindingHandler::Init()
 {
-	DeviceLayer::PlatformMgr().ScheduleWork(InitInternal);
+	InitInternal();
 }
 
-void BindingHandler::TemperatureMeasurementReadHandler(const EmberBindingTableEntry &binding,
-						       OperationalDeviceProxy *peer_device, void *context)
+void BindingHandler::OnInvokeCommandSucces(BindingData *bindingData)
 {
-	VerifyOrReturn(context != nullptr, LOG_ERR("Invalid context for Thermostat handler"););
-	BindingData *data = static_cast<BindingData *>(context);
+	LOG_DBG("Binding command applied successfully!");
 
-	TemperatureMeasurementUnicastBindingRead(data, binding, peer_device);
+	/* If session was recovered and communication works, reset flag to the initial state. */
+	if (bindingData->CaseSessionRecovered)
+		bindingData->CaseSessionRecovered = false;
 }
 
-void BindingHandler::TemperatureMeasurementUnicastBindingRead(BindingData *data, const EmberBindingTableEntry &binding,
-							      OperationalDeviceProxy *peer_device)
+void BindingHandler::OnInvokeCommandFailure(BindingData *bindingData, CHIP_ERROR aError)
```

Comment:
Let's be consistent wrt to using `aArg` notation for arguments - I would personally not use it but at least let it be consistent in a single source file :).